controllers/servingruntime_controller.go

```diff
@@ -224,6 +233,42 @@ func (r *ServingRuntimeReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{}, fmt.Errorf("Invalid runtime Spec: %w", err)
 	}
 
+	s := &corev1.Secret{}
+	if err = r.Client.Get(ctx, types.NamespacedName{
+		Name:      modelmesh.StorageSecretName,
+		Namespace: req.Namespace,
+	}, s); err != nil {
+		return ctrl.Result{}, fmt.Errorf("Could not get the storage secret: %w", err)
+	}
+
+	pvcs := make(map[string]struct{})
+	var storageConfig map[string]string
+	for _, storageData := range s.Data {
+		if err = json.Unmarshal(storageData, &storageConfig); err != nil {
+			return ctrl.Result{}, fmt.Errorf("Could not parse storage configuration json: %w", err)
+		}
+		if storageConfig["type"] == StoragePVCType {
+			pvcs[storageConfig["name"]] = struct{}{}
+		}
+	}
+	// add pvcs for predictors when the global config is enabled
+	if r.DeployPVCForPredictor {
+		list := &v1beta1.InferenceServiceList{}
+		if err = r.Client.List(ctx, list, client.InNamespace(req.Namespace)); err != nil {
+			return ctrl.Result{}, fmt.Errorf("Could not get inference services: %w", err)
+		}
+
+		for i := range list.Items {
+			isvc := &list.Items[i]
+			if isvc.Spec.Predictor.Model != nil {
+				storageUri := isvc.Spec.Predictor.Model.PredictorExtensionSpec.StorageURI
+				if u, urlErr := url.Parse(*storageUri); urlErr == nil {
+					pvcs[u.Host] = struct{}{}
+				}
+			}
+		}
```

Comment:
As discussed, change this to be a variation of [this](https://github.com/kserve/modelmesh-serving/blob/main/controllers/servingruntime_controller.go#L383) logic ... where the passed-in func is used to collect the list of PVCs. We can actually also keep the `runtimeSupportsPredictor` condition in there ... so that we only consider predictors that can actually land on this runtime.